README.md

```diff
@@ -65,9 +65,15 @@ When you have implemented the tool, answer the questions below, commit it to Git
 ## Questions
 
 How does your method for extracting features work?
+    The algorithm checks if the start position of the feature is between the positions that have been given in the query. If this is the case,
+    the chromosome is printed to the user.
 
-What is the complexity of the algorithm, as a function of the size of the two input files? When you answer this, you need to know that you can get the list of chromosomse from a `query.Table` in constant time, but it does, of course, take longer to run through all the lines in it.
+What is the complexity of the algorithm, as a function of the size of the two input files? When you answer this, you need to know that you can get the list of chromosomes from a `query.Table` in constant time, but it does, of course, take longer to run through all the lines in it.
+    The file containing the chromosomes with their features has to be parsed, and we have to run through the file containing the queries;
+    assuming that the query table is accessed in constant time, we should get a complexity of O(n+m)
 
 Did you, at any point, exploit that our features are on single nucleotides and not larger regions?
+    I've tried to avoid this, and attempted to make the algorithm as general
```

Comment:
You might have tried, but you failed. Check this line:

```python
    if queryPositionStart <= k.chrom_start < queryPositionEnd:
        ...
```

Would this work if feature regions could be larger, and we still wanted to find any features overlapping the query region?

Reply:
True, I didn't think of that. I suppose I'd need to check if any position within the feature interval is inside the query region then?

Reply:
Yes. You would want to report all features that overlap, and overlap means sharing as little as one nucleotide.

(Working with overlapping intervals is a bit of a pain, I'll tell you, but it is quite often something we need to do).